### ISSUE TITLE
Breakpoints that are set before program run cannot be deleted/disabled.

### ISSUE THREAD

**Issue by aonurK** [microsoft/vscode#146886]
Does this issue occur when all extensions are disabled?: Yes (Only the C/C++ extension is enabled and issue occured)

- VS Code Version: 1.66
- OS Version: Ubuntu 20.04.4 LTS
- GDB version: 9.2
- C/C++ Extension version: 1.9.7

When I run the program without debugging it works as expected. Also, when I set the breakpoints during application run it hits the breakpoint and after I removed the breakpoints it does not hit again. So far so good,and here are the steps to create this issue.

1. Set the breakpoint before application run
2. Debug the program,
3. It hits the breakpoint,
4. Remove the breakpoint,
5. It hits the breakpoint again!

In the breakpoints menu, as it is shown, nothing can be seen. But, when I opened the debug console and write the "-exec info break" to see current breakpoints, it shows the deleted breakpoints that is set before application run. When I manually remove it with (lets say breakpoint id is 2) "-exec d 2" it is deleted and program continues as expected. 

To sum up, it seems like there exists  unhealthy communication between vs code and gdb. The below figures better explains what I mean.    

I investigated through internet but did not find a solution for that except for manually deleting which is time consuming and annoying workaround for me. 
![1](https://user-images.githubusercontent.com/16953239/161917632-99d63933-175f-43b6-b7be-13c30355bc03.png)
![2](https://user-images.githubusercontent.com/16953239/161917636-3c172534-75f0-400c-b397-faf1e5b40805.png)
